src/main/scala/beam/agentsim/agents/ridehail/MatchmakingUtils.scala

```diff
@@ -147,25 +148,29 @@ object MatchmakingUtils {
         newPoolingList.append(temp.head)
         temp.drop(1)
     }
-    val isValid = sortedRequests.forall { curReq =>
-      val prevReq = newPoolingList.lastOption.getOrElse(newPoolingList.last)
-      val tdc = skimmer.getTimeDistanceAndCost(
-        prevReq.activity.getCoord,
-        curReq.activity.getCoord,
-        prevReq.baselineNonPooledTime,
-        BeamMode.CAR,
-        Id.create(
-          skimmer.beamScenario.beamConfig.beam.agentsim.agents.rideHail.initialization.procedural.vehicleTypeId,
-          classOf[BeamVehicleType]
+
+    var isValid = true
+    breakable {
```

Comment:
This works - although I am curious - was the code I put up not working? I wrote it in the browser, so it wasn't tested...but it seemed right :)

Reply:
yes, it is right, but it slows it down. breaking out of the loop as soon as the matching is invalid is critical.

Reply:
Interesting...it should break:

```
scala> List(true, true, false, true).forall(x=>{println(x);x})
true
true
false
res0: Boolean = false

scala> List(true, true, true, true).forall(x=>{println(x);x})
true
true
true
true
res1: Boolean = true
```